### PR TITLE
feat(@spartacus/storefront-e2e-cypress): regression/address-book to cypress (#1125)

### LINKED ISSUE #1125: migrate regression/address-book to cypress [waiting for #1216]
See #1102 for original spike.
Take a look at big happy path for example.
See cypress video in our wiki.
This one needs login so do after #1216.

### PR DESCRIPTION
**Pull request by znikola**
Closes #1125

### CODE REVIEW

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-forms.ts [hunk 1]
```diff
@@ -28,21 +28,33 @@ export interface PaymentDetails {
 export function fillShippingAddress(shippingAddressData: ShippingAddressData) {
   cy.get('cx-address-form').within(() => {
     cy.get('[formcontrolname="titleCode"]').ngSelect('Mr.');
-    cy.get('[formcontrolname="firstName"]').type(shippingAddressData.firstName);
-    cy.get('[formcontrolname="lastName"]').type(shippingAddressData.lastName);
-    cy.get('[formcontrolname="line1"]').type(shippingAddressData.address.line1);
-    cy.get('[formcontrolname="line2"]').type(shippingAddressData.address.line2);
+    cy.get('[formcontrolname="firstName"]')
```
Comment (WeizhengSap) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-forms.ts]: I guess you clear those fields for editing.

File: projects/storefrontapp-e2e-cypress/cypress/integration/address-book.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      fillShippingAddress(newAddress);
+    });
+
+    it('should display the newly added address card in the address book', () => {
+      cy.get('cx-address-card').should('have.length', 1);
+      assertAddressForm(newAddress);
+    });
+
+    it('should edit the existing address', () => {
+      cy.get('.edit').click();
```
Comment (WeizhengSap) [inline projects/storefrontapp-e2e-cypress/cypress/integration/address-book.e2e-spec.ts]: after clicking edit, we should see the form with the "newAddress" value. Then, click one of the fields to make some change.

File: projects/storefrontapp-e2e-cypress/cypress/integration/address-book.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      fillShippingAddress(editedAddress);
+    });
+
+    it('should display the edited address card in the address book', () => {
+      cy.get('cx-address-card').should('have.length', 1);
+      assertAddressForm(editedAddress);
+    });
```
Comment (WeizhengSap) [inline projects/storefrontapp-e2e-cypress/cypress/integration/address-book.e2e-spec.ts]: should add test for "add new address", which will make the page has 2 cards.

File: projects/storefrontapp-e2e-cypress/cypress/integration/address-book.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    it('should delete the existing address', () => {
+      cy.get('.delete').click();
```
Comment (WeizhengSap) [inline projects/storefrontapp-e2e-cypress/cypress/integration/address-book.e2e-spec.ts]: after delete, we first see the confirmation message, and 2 buttons. Click the "cancel" button back to original card, click the "delete" button will make the delete, we see global message and the card is gone.

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-forms.ts (hunk 1 above)
Comment (znikola) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-forms.ts]: Yes, that's correct.

File: projects/storefrontapp-e2e-cypress/cypress/integration/address-book.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+      cy.get('.cx-address-card__delete-msg').should(
+        'not.contain',
+        'Are you sure you want to delete this address?'
+      );
+
+      // click delete
+      firstCard = cy.get('cx-address-card').first();
+      firstCard.find('.delete').click();
+      cy.get('.cx-address-card--delete-mode button.btn-primary').click();
+      cy.get('cx-global-message').contains('Address deleted successfully!');
```
Comment (WeizhengSap) [inline projects/storefrontapp-e2e-cypress/cypress/integration/address-book.e2e-spec.ts]: better check that there is one card left.

**Pull request merged**